torch/nn/modules/module.py

```diff
@@ -399,2595 +404,2602 @@ def _forward_unimplemented(self, *input: Any) -> None:
+                    copied to that device
 
-        This also makes associated parameters and buffers different objects. So
-        it should be called before constructing the optimizer if the module will
-        live on MTIA while being optimized.
+            Returns:
+                Module: self
+            """
+            return self._apply(lambda t: t.cuda(device))
 
-        .. note::
-            This method modifies the module in-place.
+        def ipu(self: T, device: Optional[Union[int, device]] = None) -> T:
+            r"""Move all model parameters and buffers to the IPU.
 
-        Arguments:
-            device (int, optional): if specified, all parameters will be
-                copied to that device
+            This also makes associated parameters and buffers different objects. So
+            it should be called before constructing the optimizer if the module will
+            live on IPU while being optimized.
 
-        Returns:
-            Module: self
-        """
-        return self._apply(lambda t: t.mtia(device))
+            .. note::
+                This method modifies the module in-place.
 
-    def cpu(self: T) -> T:
-        r"""Move all model parameters and buffers to the CPU.
+            Arguments:
+                device (int, optional): if specified, all parameters will be
+                    copied to that device
 
-        .. note::
-            This method modifies the module in-place.
+            Returns:
+                Module: self
+            """
+            return self._apply(lambda t: t.ipu(device))
 
-        Returns:
-            Module: self
-        """
-        return self._apply(lambda t: t.cpu())
+        def xpu(self: T, device: Optional[Union[int, device]] = None) -> T:
+            r"""Move all model parameters and buffers to the XPU.
 
-    def type(self: T, dst_type: Union[dtype, str]) -> T:
-        r"""Casts all parameters and buffers to :attr:`dst_type`.
+            This also makes associated parameters and buffers different objects. So
+            it should be called before constructing optimizer if the module will
+            live on XPU while being optimized.
 
-        .. note::
-            This method modifies the module in-place.
+            .. note::
+                This method modifies the module in-place.
 
-        Args:
-            dst_type (type or string): the desired type
+            Arguments:
+                device (int, optional): if specified, all parameters will be
+                    copied to that device
 
-        Returns:
-            Module: self
-        """
-        return self._apply(lambda t: t.type(dst_type))
+            Returns:
+                Module: self
+            """
+            return self._apply(lambda t: t.xpu(device))
 
-    def float(self: T) -> T:
-        r"""Casts all floating point parameters and buffers to ``float`` datatype.
+        def mtia(self: T, device: Optional[Union[int, device]] = None) -> T:
```

Comment:
Perhaps to clarify why I think this doesn't really make sense: As far as I can see, these type annotations here are all pointless, because they do not appear inside the `if TYPE_CHECKER` branch. No type checker will ever care about them. We could replace, `device: Optional[Union[int, device]]` by `device: SomeGarbageAnnotation` and we will never notice. Even worse: Inside the function we could do `device.call_nonexisting_function()` and we would also never notice, because the type checker doesn't look at anything here.

And super bad for users: If we forget to move these signatures into the `if TYPE_CHECKING` block properly, it will become possible to release a pytorch version that has an incomplete/broken interface in the type stubs. After all, it might be possible to get the pytorch CI "green" by only typing a subset of the methods that are used within one of the type-checked usages inside pytorch. But if some method is never actively used, it can be omitted in the stubs, and users would get an undefined method/attribute error when trying to use an actually existing method.